tests/queries/0_stateless/03129_update_nested_materialized_column_check.sql

```diff
@@ -0,0 +1,44 @@
+SET asterisk_include_materialized_columns = 1 ;
+
+CREATE TABLE elements
+(
+    `id` UInt32,
+    `nested.key` Array(String),
+    `nested.value` Array(String),
+    `nested.key_hashed` Array(UInt64) MATERIALIZED arrayMap(x -> sipHash64(x), nested.key),
+    `nested.val_hashed` Array(UInt64) MATERIALIZED arrayMap(x -> sipHash64(x), nested.value),
+)
+    ENGINE = Memory ;
+
+
+INSERT INTO elements (id,`nested.key`,`nested.value`) VALUES (5555, ['moto', 'hello'],['chocolatine', 'croissant']);
+
+SELECT * FROM elements ;
+
+ALTER TABLE elements
+UPDATE
+
+    `nested.key` = arrayFilter(
+    (c, k, v) -> NOT (match(k, '.*') AND match(v, 'chocolatine')),
+    `nested.key`, `nested.key`, `nested.value`
+    ),
```

Comment:
In my opinion, we can simplify the `arrayFilter` condition to make the test easier to read for the reviewers. What do you think?

Reply:
sure, will adapt now 🙏